### PR TITLE
Add configuration to .travis.yml

### PR DESCRIPTION
**Pull request by aweyl**
1. added install step
2. added deploy of redisgraph to s3

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+dist: trusty
+
 env:
-- REDIS_MODULE_PATH=$TRAVIS_BUILD_DIR/src/libmodule.so REDIS_PATH=$TRAVIS_BUILD_DIR/redis/src/redis-server
+  - MODULE_NAME=redisgraph
+  - REDIS_MODULE_PATH=$TRAVIS_BUILD_DIR/src/libmodule.so
+  - REDIS_PATH=$TRAVIS_BUILD_DIR/redis/src/redis-server
+
+install:
+  - make clean && make
```
Comment (swilly22) [inline .travis.yml]: I don't see a point in running make before running cmake...

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,36 @@
...
-- make test
\ No newline at end of file
+  - cmake .
+  - make
+  - make test
+
+deploy:
+  # Module latest artifact
+  - provider: s3
+    access_key_id: $AWS_ACCESS_KEY
```
Comment (swilly22) [inline .travis.yml]: AWS_ACCESS_KEY is not defined.

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
+    secret_access_key: $AWS_SECRET
```
Comment (swilly22) [inline .travis.yml]: AWS_SECRECT isn't defined.

File: .travis.yml (hunk 1 above)
Comment (aweyl) [reply .travis.yml]: agree

Comment (aweyl) [reply .travis.yml]: My bad :)

File: .travis.yml (hunk 2 above)
Comment (aweyl) [reply .travis.yml]: The way redisearch uses access_key_id and secret_access_key in his .travis.yaml is incorrect.
The better way to use it is as it done in rejson and rebloom who take that data from those environment variables which must be declared in s3. This is why I have done it.

File: .travis.yml (hunk 3 above)
Comment (aweyl) [reply .travis.yml]: The way redisearch uses access_key_id and secret_access_key in his .travis.yaml is incorrect.
The better way to use it is as it done in rejson and rebloom who take that data from those environment variables which must be declared in s3. This is why I have done it.

**Pull request merged**